packages/app/spammer/spammer.go

```diff
@@ -83,9 +83,13 @@ func (s *Spammer) run(rate int, payloadSize uint64, timeUnit time.Duration, imif
 		case <-s.shutdown:
 			return
 		case <-ticker.C:
-			estimatedDuration := s.estimateFunc()
 			// TODO: only sleep if estimate > some threshold.
-			time.Sleep(estimatedDuration)
+			for estimatedDuration := s.estimateFunc(); estimatedDuration > 0; estimatedDuration = s.estimateFunc() {
+				if estimatedDuration > time.Duration(rate) {
+					estimatedDuration = time.Duration(rate)
+				}
```

Comment:
Use something like below instead of `if`. Return value of `min` can be also put directly into `time.Sleep`

```
estimatedDuration = lo.Min(estimatedDuration, time.Duration(rate))
```